### ISSUE TITLE
auth0-authorization-extension Rule regenerated with wrong config upon extension update

### ISSUE THREAD

**Issue by tcschiller** [auth0/auth0-authorization-extension#256]
We recently updated the Authorization extension from `2.4` to `2.6` and discovered in our custom Rules (that are called _after_ the auth0-authorization-extension Rule) that `user.permissions` was `undefined`.

I found in the source code where the template for the auth0-authorization-extension Rule lives (`server/lib/rules/authorize.js`). The template has some simple logic that outputs different lines of JavaScript based on some `config` passed in.

The Rule regenerator is called in two cases: 1) when the extension is updated, and 2) when the account configuration (e.g., permission pass-through) is changed.

Looking at the code in `server/api/extension-hooks/routes/put_update.js`, you can see `compileRule` is called with a `config` argument. `config`, though is _not_ the configuration on one's account - in this context, it's simply a hard-coded object:

```
const config = {
  groupsInToken: true,
  persistGroups: true
};
```

This means the Rule generated upon extension update will differ from the Rule that would have been generated if it used the actual configuration of one's account.

This explains why our integration broke after the upgrade, and why regenerating the Rule in the Configuration page resolved our issue.

Resolution: Update `server/api/extension-hooks/routes/put_update.js` such that it uses one's actual configuration instead of a hard-coded value.

**Comment by tcschiller** [comment]
I just received an email from Auth0 notifying us of the need to update our Auth0 extensions because of upcoming cross-domain cookie behavior changes from various browsers. Can this PR be merged before the majority of users update? Else, their integrations will break like ours did.

**Comment by karl** [comment]
I believe we have just been hit with this bug when upgrading out Auth0 Authorization extension.

It has been upgraded to 2.6 and now we no longer see roles or permissions for any of our users which is stopping anyone logging in to our site!

**Comment by karl** [comment]
For anyone else who comes across this @tcschiller mentioned the workaround in the issue description:

> regenerating the Rule in the Configuration page resolved our issue.

**Comment by ubenzer** [comment]
Hi @tcschiller , thanks for creating the issue with such a high level of detail. Due to the way hooks work with newly introduced scopes, we were unable to merge the PR as is.

With the introduction of 2.8, we stopped regenerating rules upon extension upgrade altogether. With this change, upgrading from versions >=2.6 should not cause issues anymore.

When you upgrade from versions <2.6 there are actions needed to be taken. We documented them in the [CHANGELOG](https://github.com/auth0/auth0-authorization-extension/blob/master/CHANGELOG.md).

I am closing this issue as it is solved for versions >=2.6 and we now have a migration path documented in the documentation and change log for versions <2.6.
